File: Core/Stateflows.Common/Classes/RequestResult.cs
using Newtonsoft.Json;

namespace Stateflows.Common
{
    public class RequestResult<TResponse> : SendResult
    {
        [JsonConstructor]
        protected RequestResult() : base() { }

        public RequestResult(EventHolder request, EventStatus status, EventValidation validation = null)
            : base(status, validation)
        {
            var holder = request.GetResponseHolder();
            ResponseHolder = (EventHolder<TResponse>)holder;
        }

        public RequestResult(EventHolder<TResponse> response, EventStatus status, EventValidation validation = null)
            : base(status, validation)
        {
            ResponseHolder = response;
        }

        private EventHolder<TResponse> ResponseHolder { get; set; }

        public TResponse Response => ResponseHolder != null 
            ? ResponseHolder.Payload
            : default;
    }

    public class RequestResult : SendResult
    {
        public RequestResult(EventHolder response, EventStatus status, EventValidation validation = null)
            : base(status, validation)
        {
            Response = response;
        }

        public EventHolder Response { get; set; }
    }
}

File: Transport/@stateflows/signalr-client/src/classes/request-result.ts
import { EventStatus } from "../enums/event-status";
import { EventValidation } from "./event-validation";
import { SendResult } from "./send-result";

export class RequestResult<TResponse> extends SendResult {
    constructor(
        public Response: TResponse,
        event: Event,
        status: EventStatus,
        validation: EventValidation,
    ) {
        super(event, status, validation);
    }
}